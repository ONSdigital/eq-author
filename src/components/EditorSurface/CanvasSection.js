import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { colors, shadow } from "constants/theme";
import { transparentize } from "polished";

const focusedStyle = css`
  box-shadow: none;
  outline-color: ${colors.lightBlue} !important;
`;

const StyledCanvasSection = styled.div`
  padding: 2em 2.5em;
  transition: outline-color 100ms ease-in;
  background: white;
  margin-bottom: 1px;
  position: relative;
  outline-width: 1px;
  outline-style: solid;
  outline-color: transparent;
  box-shadow: ${shadow};

  &:hover {
    outline-color: ${transparentize(0.6, colors.lightBlue)};
  }
  ${props => props.isFocused && focusedStyle};
`;

export default class CanvasSection extends React.Component {
  static propTypes = {
    children: PropTypes.node,
    onFocus: PropTypes.func.isRequired,
    isFocused: PropTypes.bool,
    id: PropTypes.string.isRequired
  };

  static defaultProps = {
    isFocused: false
  };

  handleFocus = () => {
    this.props.onFocus(this.props.id);
  };

  render() {
    return (
      <StyledCanvasSection
        {...this.props}
        onClick={this.handleFocus}
        onFocus={this.handleFocus}
      />
    );
  }
}
