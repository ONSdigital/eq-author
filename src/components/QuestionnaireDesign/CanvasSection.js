import React, { cloneElement, Component, Children } from "react";
import PropTypes from "prop-types";
import { noop } from "lodash";
import styled from "styled-components";
import { colors } from "constants/theme";
import { transparentize } from "polished";

const StyledCanvasSection = styled.div`
  padding: 2em 2.625em;
  margin-bottom: -2px;
  border-bottom: 2px dashed #c6c6c6;
  outline-offset: -2px;
  outline: ${props =>
    props.focused ? `2px solid ${colors.lightBlue} !important` : "none"};

  &:last-child {
    border: none;
  }
  &:hover {
    outline: 2px solid ${transparentize(0.6, colors.lightBlue)};
  }
`;

export default class CanvasSection extends Component {
  static propTypes = {
    children: PropTypes.node,
    onFocus: PropTypes.func,
    focused: PropTypes.bool,
    id: PropTypes.string
  };

  static defaultProps = {
    onFocus: noop,
    onBlur: noop,
    focused: false
  };

  handleFocus = e => {
    this.props.onFocus(this.props.id);
  };

  render() {
    const { children, focused } = this.props;

    return (
      <StyledCanvasSection onFocus={this.handleFocus} focused={focused}>
        {Children.map(children, child => cloneElement(child, { focused }))}
      </StyledCanvasSection>
    );
  }
}
