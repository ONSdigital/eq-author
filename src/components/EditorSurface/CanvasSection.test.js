import React from "react";
import { shallow } from "enzyme";
import CanvasSection from "./CanvasSection";

let component, handleFocus, handleBlur;

const Child = () => <div />;

describe("CanvasSection", () => {
  beforeEach(() => {
    handleFocus = jest.fn();
    handleBlur = jest.fn();

    component = shallow(
      <CanvasSection id="foo" onFocus={handleFocus} onBlur={handleBlur} focused>
        <Child />
      </CanvasSection>
    );
  });

  it("should render", () => {
    expect(component).toMatchSnapshot();
  });

  it("should handle focus event", () => {
    component.simulate("focus");
    expect(handleFocus).toHaveBeenCalledWith("foo");
  });

  it("should handle blur", () => {
    component.simulate("blur");
    expect(handleBlur).toHaveBeenCalled();
  });
});
