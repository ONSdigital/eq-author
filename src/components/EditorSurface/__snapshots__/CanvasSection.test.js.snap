// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasSection should render 1`] = `
<CanvasSection__FocusableSection
  focusOnSection={[Function]}
  id="foo"
  isFocused={true}
  onBlur={[Function]}
  onFocus={[Function]}
>
  <Child />
</CanvasSection__FocusableSection>
`;
